snuba/consumers/consumer_builder.py

```diff
@@ -55,19 +56,32 @@ def __init__(
             self.bootstrap_servers = bootstrap_servers
 
         stream_loader = enforce_table_writer(self.dataset).get_stream_loader()
-        self.raw_topic = raw_topic or stream_loader.get_default_topic_spec().topic_name
-        default_replacement_topic_name = (
-            stream_loader.get_replacement_topic_spec().topic_name
-            if stream_loader.get_replacement_topic_spec()
-            else None
-        )
-        self.replacements_topic = replacements_topic or default_replacement_topic_name
-        default_commit_log_topic_name = (
-            stream_loader.get_commit_log_topic_spec().topic_name
-            if stream_loader.get_commit_log_topic_spec()
-            else None
-        )
-        self.commit_log_topic = commit_log_topic or default_commit_log_topic_name
+
+        self.raw_topic: Topic
```

Comment:
This is nice because it defines but doesn't initialize the variable, like in this example:

```python
class Thing:
    def __init__(self) -> None:
        self.name: str

t = Thing()
t.name
```

```
ted@veneno % python /tmp/test.py
Traceback (most recent call last):
  File "/tmp/test.py", line 6, in <module>
    t.name
AttributeError: 'Thing' object has no attribute 'name'
```

I'd prefer to do this assignment in a single expression but I couldn't think of a clean way to handle the optional topics (replacements, commit log.)

Reply:
But do you actually need it in this specific case ?
both the branches below will assign type Topic to self.raw_topic. Line 61 to 65 should be equivalent to this one for the typechecker.

Reply:
Specifically for `raw_topic` — no, they're both non-optional. I mostly did it for consistency with the subsequent assignments (which do need it.) I'm happy to remove it if you think it's cleaner without.

Reply:
no, no need to remove it, I don't think it would be clearer. Now I get why it is needed.
I think the issue you are running into without explicitly defining the type is that the typechecker cannot guarantee that any other method may have assigned something of a different type to self.raw_topic between your assignment and the line where you pass it to the next function. Thus it makes sense to establish the type in the constructor.